### PR TITLE
Albert/line kick

### PR DESCRIPTION
**Pull request by ashaw596**
Implements LineKick Planner. Adds debug text to Paths. Increases font size to prevent squinting.

### CODE REVIEW

**Comment by justbuchanan** [comment]
I got a segfault when trying to run `TestLineKick`:

~~~{.sh}
Thread 8 "QThread" received signal SIGABRT, Aborted.
[Switching to Thread 0x7fffcb7fe700 (LWP 15571)]
0x00007ffff2aca295 in raise () from /usr/lib/libc.so.6
(gdb) bt
#0  0x00007ffff2aca295 in raise () from /usr/lib/libc.so.6
#1  0x00007ffff2acb6da in abort () from /usr/lib/libc.so.6
#2  0x00007ffff33e33dd in __gnu_cxx::__verbose_terminate_handler () at /build/gcc/src/gcc/libstdc++-v3/libsupc++/vterminate.cc:95
#3  0x00007ffff33e1196 in __cxxabiv1::__terminate (handler=<optimized out>) at /build/gcc/src/gcc/libstdc++-v3/libsupc++/eh_terminate.cc:47
#4  0x00007ffff33e11e1 in std::terminate () at /build/gcc/src/gcc/libstdc++-v3/libsupc++/eh_terminate.cc:57
#5  0x00007ffff33e13f8 in __cxxabiv1::__cxa_throw (obj=0x7fffbc492a10, tinfo=0x7ffff36ca830 <typeinfo for std::bad_cast>, dest=0x7ffff33df670 <std::bad_cast::~bad_cast()>)
    at /build/gcc/src/gcc/libstdc++-v3/libsupc++/eh_throw.cc:87
#6  0x00007ffff33e0062 in __cxxabiv1::__cxa_bad_cast () at /build/gcc/src/gcc/libstdc++-v3/libsupc++/eh_aux_runtime.cc:34
#7  0x00007ffff7971137 in Planning::RRTPlanner::run (this=0x7fffbc4937f0, planRequest=...) at ../soccer/planning/RRTPlanner.cpp:80
#8  0x00007ffff799c8e6 in Planning::LineKickPlanner::run (this=0x7fffbc4937e0, planRequest=...) at ../soccer/planning/LineKickPlanner.cpp:94
#9  0x00007ffff7964a70 in Planning::IndependentMultiRobotPathPlanner::run (this=0xb71c60, requests=std::map with 6 elements = {...})
    at ../soccer/planning/IndependentMultiRobotPathPlanner.cpp:78
#10 0x00007ffff79a8463 in Processor::run (this=0x692be0) at ../soccer/Processor.cpp:468
#11 0x00007ffff5245b38 in ?? () from /usr/lib/libQt5Core.so.5
#12 0x00007ffff404b484 in start_thread () from /usr/lib/libpthread.so.0
#13 0x00007ffff2b7e6dd in clone () from /usr/lib/libc.so.6
~~~

File: soccer/planning/IndependentMultiRobotPathPlanner.cpp [hunk 1]
```diff
@@ -37,7 +37,7 @@ std::map<int, std::unique_ptr<Path>> IndependentMultiRobotPathPlanner::run(
 
     // Sorts descending so that higher priorities are first
     auto comparator = [&](int& lhs, int& rhs) {
-        return requests[lhs].priority > requests[rhs].priority;
+        return requests.at(lhs).priority > requests.at(rhs).priority;
```
Comment (justbuchanan) [inline soccer/planning/IndependentMultiRobotPathPlanner.cpp]: Any reason for `at()` instead of `[]`?

**Comment by ashaw596** [comment]
How are you getting that error? Thats a cast error for a command.

    assert(planRequest.cmd.getCommandType() ==
           Planning::MotionCommand::PathTarget);
    const Planning::PathTargetCommand& target =
        dynamic_cast<const Planning::PathTargetCommand&>(planRequest.cmd);

After that assert that should be impossible?? Are you doing anything in particular? I can't replicate on my mac.

File: soccer/planning/IndependentMultiRobotPathPlanner.cpp (hunk 1 above)
Comment (ashaw596) [inline soccer/planning/IndependentMultiRobotPathPlanner.cpp]: At checks bounds and throws an error. Also doesn't create a new object if it doesn't exist so it is const.

**Comment by jgkamat** [comment]
@ashaw596, since this is in comp2016, can we close this merge?

**Pull request closed without merging**